workflows/pr_gate/Jenkinsfile

```diff
@@ -0,0 +1,99 @@
+@Library('my_library') _
+node{
+    timestamps{
+        withEnv([
+            "PR_URL=${env.ghprbPullLink}",
+            "PR_TARGET_BRANCH=${env.ghprbTargetBranch}",
+            "PR_ID=${env.ghprbPullId}",
+            "PR_REPO=${env.ghprbGhRepository}",
+            "PUBLIC_JENKINS_URL=${env.PUBLIC_JENKINS_URL}",
+            "TEST_STACK=${env.TEST_STACK}",
+            "TESTS=${env.TESTS}",
+            "KEEP_FAILURE_ENV=${env.KEEP_FAILURE_ENV}",
+            "KEEP_MINUTES=${env.KEEP_MINUTES}",
+            "KEEP_DOCKER_ON_FAILURE=${env.KEEP_DOCKER_ON_FAILURE}"
+        ]){
+            deleteDir() 
+            def manifest = new pipeline.common.Manifest()
+            def share_method = new pipeline.common.ShareMethod()
+            String manifest_path = "manifest_$PR_ID"
+            String library_dir = "on-build-config"
+            def manifest_dict = null
+            
+            try{
+                stage("Parse Pull Request"){
+                    share_method.checkoutOnBuildConfig(library_dir)
+                    manifest_path = manifest.generateManifestFromPR(PR_URL, PR_TARGET_BRANCH, manifest_path, library_dir, "JENKINSRHD_GITHUB_TOKEN")
+                    archiveArtifacts "$manifest_path"
+                    manifest_dict = manifest.stashManifest("manifest", manifest_path)
+                }
+                stage("Unit Test"){
+                    ArrayList<String> used_resources = []
+                    def test_repos = manifest.getReposNeedTest(manifest_path, library_dir)
+                    def unit_test = new pipeline.rackhd.source_code.UnitTest()
+                    def unit_test_branches = [:]
+                    for(int i=0; i<test_repos.size; i++){
+                        String repo_name = test_repos.get(i)
+                        unit_test_branches["${repo_name}"] = {
+                            if(repo_name == "on-http" || repo_name == "on-core"){
+                                unit_test.runTestWithSudo(repo_name, manifest_dict, used_resources)
```

Comment:
can you add comment that "because their unit-test requires mongodb service start/stop ?"

Reply:
Thanks, fixed!